cpp/models/ode_secirts/parameters_io.h

```diff
@@ -418,7 +413,12 @@ set_confirmed_cases_data(std::vector<Model>& model, const std::vector<ConfirmedC
                 immunity_population[0][i] * denom_I_Sy[i] * num_InfectedSymptoms[county][i];
             model[county].populations[{AgeGroup(i), InfectionState::InfectedSymptomsNaiveConfirmed}] = 0;
             model[county].populations[{AgeGroup(i), InfectionState::InfectedSevereNaive}] =
-                immunity_population[0][i] * denom_I_Sev[i] * num_InfectedSevere[county][i];
+                immunity_population[0][i] * denom_I_Sev_Cr[i] * num_InfectedSevere[county][i];
+            // Only set the number of ICU patients here, if the date is not available in the data.
+            if (date <= Date(2020, 4, 23) || date >= Date(2024, 7, 21)) {
+                model[county].populations[{AgeGroup(i), InfectionState::InfectedCriticalNaive}] =
+                    immunity_population[0][i] * denom_I_Sev_Cr[i] * num_icu[county][i];
+            }
```

Comment:
Why are there three new computations in this function? Were they missing before?

Reply:
As the read_icu was hardcoded before, there was no method to set the icu data if it was not included in the DIVI data. So, we needed to add this here.